src/server/server.ts

```diff
@@ -48,6 +53,56 @@ function isOfficeHost(host: string) {
 }
 
 
+let generatedDirectories = {};
+/* Cleans up generated template directories */
+setInterval(() => {
+    let ids = Object.keys(generatedDirectories);
+    for (let id of ids) {
+        let directoryInfo = generatedDirectories[id];
+        let now = (new Date()).getTime();
+        // If generated file is more than an hour old and isn't being read, delete it from both server and map to clean up clutter
+        if (directoryInfo && !directoryInfo.isBeingRead && (now - directoryInfo.timestamp) > 3600000) {
+            delete generatedDirectories[id];
+            fs.unlink(directoryInfo.name, (err) => {
+                if (err) {
+                    ai.trackEvent('File deletion failed', { name: directoryInfo.name });
+                    return;
+                }
+            });
+        }
+    }
+}, 3600000);
+
+/* Cleans up lurker template directories that may not have been deleted by above function (perhaps due to server restart) */
+setInterval(() => {
+    let filenames: string[] = [];
+    for (let id of Object.keys(generatedDirectories)) {
+        filenames.push(generatedDirectories[id].relativeFilePath);
+    }
+
+    fs.readdir(path.resolve(__dirname, 'working'), (err, files) => {
+        if (err) {
+            ai.trackEvent('Error reading directory');
+            return;
+        }
+
+        files.forEach(file => {
+            let relativePath = `working/${file}`;
+            let absolutePath = path.resolve(__dirname, relativePath);
+            fs.stat(absolutePath, (err, stat) => {
+                let now = (new Date()).getTime();
+
+                // Delete all directories older than three hours and all files that are not in map
+                if (stat.isDirectory() && (now - stat.mtime.getMilliseconds()) > 10800000) {
+                    rimraf(absolutePath, () => {});
+                } else if (filenames.indexOf(relativePath) < 0 && (now - stat.mtime.getMilliseconds()) > 10800000) {
+                    fs.unlink(absolutePath);
+                }
+            });
+        });
+    });
+}, 10800000);
```

Comment:
Ditto for this constant (define at top of function and describe what the number means)